### ISSUE TITLE
SFML not compiling on windows 10 with new 0.16.0

### ISSUE THREAD

**Issue by jcbritobr** [jeremyletang/rust-sfml#235]
Tryed to compile some samples on windows 10 x64 with the new 0.16.0 and got the following stack trace:

```
w" "-lcsfml-audio" "-lcsfml-system" "-ladvapi32" "-lws2_32" "-luserenv" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\jcbri\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:/Bin/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Workspace\rust\sfml\target\debug\deps\libsfml-630982b2ac42bd62.rlib(sfml-630982b2ac42bd62.sfml.325wk4ff-cgu.15.rcgu.o): in function `sfml::graphics::render_texture::RenderTexture::with_settings':
          C:\Users\jcbri\.cargo\registry\src\github.com-1ecc6299db9ec823\sfml-0.16.0\src\graphics\render_texture.rs:57: undefined reference to `sfRenderTexture_createWithSettings'
          collect2.exe: error: ld returned 1 exit status
``´

This happens on both msvc and gnu toolchains.

**Comment by crumblingstatue** [comment]
What version of CSFML do you have?

**Comment by jcbritobr** [comment]
I'm using with msys2. 

mingw64/mingw-w64-x86_64-csfml 2.5-1 [installed]
    A simple, fast, cross-platform, and object-oriented multimedia API for C
    (mingw-w64)

Also, I installed rust sfml 0.15.1 and some sample worked. The only thats not workin was the vertex_buffer, but because there are no implementations of vertexbuffer in 0.15.1

**Comment by crumblingstatue** [comment]
It could be possible that the mingw-w64 build of CSFML somehow lacks that symbol. I tried with the official CSFML download (https://www.sfml-dev.org/download/csfml/), and I can cross-compile to windows successfully.
I suggest you try calling `sfRenderTexture_createWithSettings` with a simple test C program. If it doesn't work, the problem is with the version of CSFML that mingw-w64 ships. If it works, it's much more likely to be a problem with rust-sfml.

**Comment by jcbritobr** [comment]
> 
> 
> It could be possible that the mingw-w64 build of CSFML somehow lacks that symbol. I tried with the official CSFML download (https://www.sfml-dev.org/download/csfml/), and I can cross-compile to windows successfully.
> I suggest you try calling `sfRenderTexture_createWithSettings` with a simple test C program. If it doesn't work, the problem is with the version of CSFML that mingw-w64 ships. If it works, it's much more likely to be a problem with rust-sfml.

Already testes with the msvc toolchain and the downloaded package fom sfml page.
They are working correctly with the 0.15.1

**Comment by crumblingstatue** [comment]
What kind of programs are you trying to build?
I tried building the pong example for windows using 0.16.0, and it worked.

**Comment by jcbritobr** [comment]
> 
> 
> What kind of programs are you trying to build?
> I tried building the pong example for windows using 0.16.0, and it worked.

I was trying to use vertex_buffer. Unfortulatelly cant use it in 0.15.1, but all the other stuffs works very well.

**Comment by crumblingstatue** [comment]
The problem seems to be that CSFML still hasn't released a new version ever since https://github.com/SFML/CSFML/pull/137.
I'll try asking for a new release. In the meantime, you could build CSFML from the git master branch.

I might also try to experiment vendoring CSFML, and building it from source when building SFML. And maybe try streamlining it and tailoring it for rust-sfml.

I don't fully understand why you're having linking issues though if none of your code uses [RenderTexture::with_settings](https://docs.rs/sfml/0.16.0/sfml/graphics/struct.RenderTexture.html#method.with_settings). It could be how libraries are handled on windows. I can't reproduce, because cross-compiling from linux to windows seems to work.

**Comment by crumblingstatue** [comment]
@jcbritobr If possible, could you try out the latest version from the `vendored-csfml` branch?
You need to download the right version of SFML for your platform, and set the `SFML_INCLUDE_DIR` environment variable to where the includes are, and `SFML_LIBS_DIR` to where the libraries are, and it should build hopefully.

**Comment by lza-official** [comment]
@crumblingstatue I come across the same problem and I checked the signature of sfRenderTexture_createWithSettings.
The author of CSFML has changed its signature, which may be the reason.
Could you please correct it in a new release? I wil definitely appreciate it.

**Comment by crumblingstatue** [comment]
@lza-official See the linked issues. The problem is on the CSFML side, and it has already been corrected. The problem is that a new release with that fix hasn't been made by the CSFML devs.

**Comment by jcbritobr** [comment]
> 
> 
> @lza-official See the linked issues. The problem is on the CSFML side, and it has already been corrected. The problem is that a new release with that fix hasn't been made by the CSFML devs.

Hope the guys do some updates. Maybe in msys2.

**Comment by crumblingstatue** [comment]
This problem should be resolved when using CSFML 2.5.1, but feel free to reopen if that's not the case.
